homeassistant/components/axis.py

```diff
@@ -0,0 +1,347 @@
+                _LOGGER.error('Couldn\'t set up %s', config['name'])
+
+    return True
+
+
+def setup_device(config):
+    """Set up device."""
+    from axis import AxisDevice
+
+    hass = config['hass']
+    device = AxisDevice(config)  # Initialize device
+    enable_metadatastream = False
+
+    if device.serial_number is None:
+        # If there is no serial number a connection could not be made
+        _LOGGER.error('Couldn\'t connect to %s', config[CONF_HOST])
+        return False
+
+    for component in config[CONF_INCLUDE]:
+        if component in EVENT_TYPES:
+            # Sensors are created by device calling event_initialized
+            # when receiving initialize messages on metadatastream
+            device.add_event_topic(convert(component, 'type', 'subscribe'))
+            if not enable_metadatastream:
+                enable_metadatastream = True
+        else:
+            discovery.load_platform(hass, component, DOMAIN, config)
+
+    if enable_metadatastream:
+        device.initialize_new_event = event_initialized
+        device.initiate_metadatastream()
+    AXIS_DEVICES[device.serial_number] = device
+    return True
+
+
+def _read_config(hass):
+    """Read Axis config."""
+    path = hass.config.path(CONFIG_FILE)
+
+    if not os.path.isfile(path):
+        return {}
+
+    with open(path) as f_handle:
+        # Guard against empty file
+        return json.loads(f_handle.read() or '{}')
+
+
+def _write_config(hass, config):
+    """Write Axis config."""
+    data = json.dumps(config)
+    with open(hass.config.path(CONFIG_FILE), 'w', encoding='utf-8') as outfile:
+        outfile.write(data)
+
+
+def event_initialized(event):
+    """Register event initialized on metadatastream here."""
+    hass = event.device_config('hass')
+    discovery.load_platform(hass,
+                            convert(event.topic, 'topic', 'platform'),
+                            DOMAIN, {'axis_event': event})
+
+
+class AxisDeviceEvent(Entity):
+    """Representation of a Axis device event."""
+
+    def __init__(self, axis_event):
+        """Initialize the event."""
+        self.axis_event = axis_event
+        self._event_class = convert(self.axis_event.topic, 'topic', 'class')
+        self._name = '{}_{}_{}'.format(self.axis_event.device_name,
+                                       convert(self.axis_event.topic,
+                                               'topic', 'type'),
+                                       self.axis_event.id)
+        self._delay = self.axis_event.device_config(CONF_SCAN_INTERVAL)
+        self._timer = None
+        self.axis_event.callback = self._update_callback
+        self.update()
+
+    def _update_callback(self):
+        """Update the sensor's state, if needed."""
+        self.update()
+
+        if self._timer is not None:
+            self._timer()
+            self._timer = None
+
+        if self._delay > 0 and not self.is_on:  # pylint: disable=E1101
```

Comment:
Don't disable this message. Instead add an `is_on` property that raises `NotImplementedError`